march_hardware/src/ethercat/ethercat_master.cpp

```diff
@@ -196,8 +203,11 @@ void EthercatMaster::ethercatLoop()
 
     if (slave_lost_duration > slave_watchdog_timeout)
     {
-      throw error::HardwareException(error::ErrorType::SLAVE_LOST_TIMOUT, "Slave connection lost for %i ms",
-                                     this->slave_watchdog_timeout_);
+      this->last_exception_ = std::make_exception_ptr(error::HardwareException(
+          error::ErrorType::SLAVE_LOST_TIMOUT, "Slave connection lost for %i ms", this->slave_watchdog_timeout_));
```

Comment:
Why do you need to do it like this instead of just directly throwing the error here?

Reply:
Since there is no place to catch this, because it is in a separate thread. This enables us to read the exception.